tests/infratests/akoinfra_test.go

```diff
@@ -679,6 +716,159 @@ func TestAKOInfraMultiAviInfraSettingCreationVPC(t *testing.T) {
+		} else {
+			return *infraSetting.Spec.NSXSettings.T1LR == "/orgs/default/projects/test-project/vpcs/"+gatewayPathName &&
+				len(infraSetting.Spec.Network.VipNetworks) == 0 &&
+				infraSetting.Spec.SeGroup.Name == "AVISEG2"
+		}
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	g.Eventually(func() bool {
+		if ns, err := kubeClient.CoreV1().Namespaces().Get(context.TODO(), "default", metav1.GetOptions{}); err != nil {
+			return false
+		} else {
+			return ns.Annotations[lib.InfraSettingNameAnnotation] == infraSettingName2 &&
+				ns.Annotations[lib.TenantAnnotation] == "test-project"
+		}
+	}, 10*time.Second).Should(gomega.Equal(true))
+
+	g.Eventually(func() bool {
+		if _, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().AviInfraSettings().Get(context.TODO(), infraSettingName, metav1.GetOptions{}); err != nil {
+			return false
+		}
+		return true
+	}, 10*time.Second).Should(gomega.Equal(false))
+
+	annotateNamespaceWithSEG(t, "default", "")
+	infraSettingName3 := lib.GetAviInfraSettingName("test-project" + gatewayPathName)
+
+	g.Eventually(func() bool {
+		if infraSetting, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().AviInfraSettings().Get(context.TODO(), infraSettingName3, metav1.GetOptions{}); err != nil {
+			return false
+		} else {
+			return *infraSetting.Spec.NSXSettings.T1LR == "/orgs/default/projects/test-project/vpcs/"+gatewayPathName &&
+				len(infraSetting.Spec.Network.VipNetworks) == 0 &&
+				infraSetting.Spec.SeGroup.Name == "Default-Group"
+		}
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	g.Eventually(func() bool {
+		if ns, err := kubeClient.CoreV1().Namespaces().Get(context.TODO(), "default", metav1.GetOptions{}); err != nil {
+			return false
+		} else {
+			return ns.Annotations[lib.InfraSettingNameAnnotation] == infraSettingName3 &&
+				ns.Annotations[lib.TenantAnnotation] == "test-project"
+		}
+	}, 10*time.Second).Should(gomega.Equal(true))
+
+	g.Eventually(func() bool {
+		if _, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().AviInfraSettings().Get(context.TODO(), infraSettingName2, metav1.GetOptions{}); err != nil {
+			return false
+		}
+		return true
+	}, 10*time.Second).Should(gomega.Equal(false))
+
+	err = dynamicClient.Resource(lib.VPCNetworkConfigurationGVR).Delete(context.TODO(), vpcNetConfig, v1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("failed to delete namespacenetworkinfo CR, error: %s", err.Error())
+	}
+
+	g.Eventually(func() bool {
+		if _, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().AviInfraSettings().Get(context.TODO(), infraSettingName3, metav1.GetOptions{}); err != nil {
+			return false
+		}
+		return true
+	}, 10*time.Second).Should(gomega.Equal(false))
+
+	g.Eventually(func() bool {
+		if ns, err := kubeClient.CoreV1().Namespaces().Get(context.TODO(), "default", metav1.GetOptions{}); err != nil {
+			return false
+		} else {
+			return ns.Annotations[lib.InfraSettingNameAnnotation] != infraSettingName
```

Comment:
I think you meant infraSettingName3, but the correct check would be for the Key itself not being present.